alibi/explainers/anchors/anchor_base.py

```diff
@@ -790,8 +790,12 @@ def anchor_beam(self, delta: float = 0.05, epsilon: float = 0.1, desired_confide
         # if no result is found, choose highest precision of best result candidate from every round
         if not best_anchor:
             success = False  # indicates the method has not found an anchor
-            logger.warning('Could not find an result satisfying the {} precision constraint. Now returning '
-                           'the best non-eligible result.'.format(desired_confidence))
+            logger.warning(f'Could not find an anchor satisfying the {desired_confidence} precision constraint. '
+                           f'Now returning the best non-eligible result. The desired precision threshold might not be '
+                           f'achieved due to the quantile-based discretisation of the numerical features. The '
+                           f'resolution of the bins may be too large to find an anchor of required precision. '
+                           f'Note that higher resolution may or may not be easily achieved depending on the underling '
```

Comment:
underling -> underlying

With respect to the message itself, should we replace the last sentence with something actionable? E.g. mention increasing the number of bins in `disc_perc` (whilst keeping the caveat that it may not help if the numerical feature distributions are skewed).